specification/synapse/data-plane/Microsoft.Synapse/preview/2021-07-01-preview/artifacts.json

```
@@ -0,0 +1,5064 @@
+        "parameters": [
+          {
+            "$ref": "#/parameters/NotebookName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "request",
+            "description": "proposed new name.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ArtifactRenameRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK."
+          },
+          "202": {
+            "description": "Accepted."
+          },
+          "default": {
+            "description": "An error response received from the Azure Synapse service.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/kqlScripts": {
+      "get": {
+        "tags": [
+          "kqlScripts"
+        ],
+        "operationId": "KqlScripts_GetAll",
+        "description": "Get all KQL scripts",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/KqlScripts_ResourceCollectionResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json#/definitions/ErrorContract"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "KqlScripts_GetAll": {
+            "$ref": "./examples/KqlScripts_GetAll.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/kqlScripts/{kqlScriptName}": {
+      "put": {
+        "tags": [
+          "kqlScripts"
+        ],
+        "operationId": "KqlScript_CreateOrUpdate",
+        "description": "Creates or updates a KQL Script",
+        "parameters": [
+          {
+            "$ref": "#/parameters/KqlScriptName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "kqlScript",
+            "description": "KQL script",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/KqlScriptResource"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/KqlScriptResource"
+            }
+          },
+          "202": {
+            "description": "Accepted"
+          },
```

A 202 response should include an `Operation-Location` response header.
```suggestion
            "headers": {
                "Operation-Location": {
                  "type": "string",
                  "description": "The URL that will return the operation result"
                }
            }
          },
```

---

This requires server side to return "Operation-Location" header